src/RadPulse/test_radiation_pulse.cpp

```diff
@@ -63,22 +65,33 @@ auto compute_exact_Trad(const double x, const double t) -> double
 	return 0.5 * normfac * std::exp(-(x * x) / (4.0 * width_sq));
 }
 
-template <> AMREX_GPU_HOST_DEVICE auto RadSystem<PulseProblem>::ComputePlanckOpacity(const double rho, const double Tgas) -> double
+template <> AMREX_GPU_HOST_DEVICE auto RadSystem<PulseProblem>::ComputePlanckOpacity(const double rho, const double Tgas) -> quokka::valarray<double, nGroups_>
 {
-	return (kappa0 / rho) * std::max(std::pow(Tgas / T0, 3), 1.0);
+	quokka::valarray<double, nGroups_> kappaPVec{};
+	auto kappa = (kappa0 / rho) * std::max(std::pow(Tgas / T0, 3), 1.0);
+	for (int i = 0; i < nGroups_; ++i) {
+		kappaPVec[i] = kappa;
+	}
+	return kappaPVec;
 }
 
-template <> AMREX_GPU_HOST_DEVICE auto RadSystem<PulseProblem>::ComputeRosselandOpacity(const double rho, const double Tgas) -> double
+template <>
+AMREX_GPU_HOST_DEVICE auto RadSystem<PulseProblem>::ComputeFluxMeanOpacity(const double rho, const double Tgas) -> quokka::valarray<double, nGroups_>
 {
-	return (kappa0 / rho) * std::max(std::pow(Tgas / T0, 3), 1.0);
+	return ComputePlanckOpacity(rho, Tgas);
 }
 
-template <> AMREX_GPU_HOST_DEVICE auto RadSystem<PulseProblem>::ComputePlanckOpacityTempDerivative(const double rho, const double Tgas) -> double
+template <>
+AMREX_GPU_HOST_DEVICE auto RadSystem<PulseProblem>::ComputePlanckOpacityTempDerivative(const double rho, const double Tgas)
+    -> quokka::valarray<double, nGroups_>
 {
+	quokka::valarray<double, nGroups_> opacity_deriv{};
+	double opacity_deriv_scalar = 0.;
 	if (Tgas > 1.0) {
-		return (kappa0 / rho) * (3.0 / T0) * std::pow(Tgas / T0, 2);
+		opacity_deriv_scalar = (kappa0 / rho) * (3.0 / T0) * std::pow(Tgas / T0, 2);
 	}
-	return 0.;
+	opacity_deriv.fillin(opacity_deriv_scalar);
```

Comment:
Does `fillin` work on GPU?

Reply:
It works on variables that are defined locally or hand-written number like 0.0. It doesn't work on globally defined constexpr like kappa0.